### DISCUSSION TITLE
zfs --version output

### DISCUSSION THREAD

**Discussion by whoschek** [openzfsonwindows/openzfs#419] [Development]
Can someone please show me an example output blurb of zfs --version on Windows on their machine. I'm trying to figure out how best to detect if the zfs version on the remote side is running on windows, as opposed to another OS. Thanks!

**Comment by guenther-alka** [comment]
zfs/zpool version gives a reference ex to 2.2.6 or 
--version zfs-2.1.99-3126-g97bdecb6f

If you need to detect if ZFS is running under Windows, use 
```zfs get driveletter```

On Windows you get the driveletter value, on another platform an error "invalid property"

**Comment by whoschek** [comment]
Ok, but does zfs --version by any chance contain "winzfs" or some unique blurb like that?

**Comment by lundman** [comment]
```
Microsoft Windows [Version 10.0.22621.4317]
(c) Microsoft Corporation. All rights reserved.

C:\Windows\System32>zpool version
zfswin-2.2.6rc10
zfs-kmod-zfswin-2.2.6rc10

C:\Windows\System32>
```

**Comment by sskras** [comment]
The `--version` format works too:
```cmd
Microsoft Windows [Version 10.0.19044.3086]
(c) Microsoft Corporation. All rights reserved.

C:\Users\saukrs> sudo zfs --version
zfswin-2.2.3rc5
zfs-kmod-zfswin-2.2.3rc5

C:\Users\saukrs>
```

**Reply by sskras** [reply]
I used `sudo` (from **gsudo** package) because `zfs.exe` requires elevated privileges by default.

**Reply by lundman** [reply]
How do you find sudo? Works well? Supports sudoers file and all that jazz?

**Reply by whoschek** [reply]
Thanks! zfswin is a great unique blurb to work with.
But wow, even something as harmless as the --version flag requires elevated privileges? That's surprising to me.

**Reply by sskras** [reply]
> How do you find sudo? Works well? Supports sudoers file and all that jazz?

@lundman, I just use it to avoid the need to repeat all my commands in the elevated, newly launched console/terminal window.
For that **gsudo** works just fine. Haven't tried other features / other implementations.

**Reply by lundman** [reply]
> But wow, even something as harmless as the --version flag requires elevated privileges? That's surprising to me.

The .exe gets ticked to need elevated privileges, so we don't get a chance to have an opinion about it. There probably are things that can be done about it, but fairly low priority so far.

**Comment by whoschek** [comment]
BTW, I have no idea if this works, as last time I used windows was ages ago, and I really don't know my way round there anymore (please be kind). But just in case it might be useful to someone, I made some tentative changes to bzfs to (maybe) make snapshot replication from windows to Unix work with zfs on windows. It uses that zfswin blurb as identifier and runs zfs.exe with sudo if that's available per the comment above. See this experimental branch here: https://github.com/whoschek/bzfs/tree/wip/zfswin

Give it a shot if you like, if only to test the new zfs send/receive bits on windows. You'd need to run the initiator on a Unix machine, and have an openssh server on windows, but with some luck it might actually work.
